src/resolve/resolved-dns-transaction.c

```diff
@@ -2850,6 +2850,43 @@ static int dns_transaction_validate_dnskey_by_ds(DnsTransaction *t) {
         return 0;
 }
 
+typedef enum DelegationType {
+        DELEGATION_NONE,
+        DELEGATION_SECURE,
+        DELEGATION_INSECURE,
+} DelegationType;
+
+/* Evaluate the delegation type expressed in a transaction */
+static int dns_delegation_type(DnsTransaction *t) {
+        _cleanup_(dns_resource_key_unrefp) DnsResourceKey *ns = NULL;
+        int r;
+        DnssecNsecResult nsec_result;
+
+        assert(t);
+
+        /* Check that this is a answer to a query for a DS record */
+        if (dns_transaction_key(t)->type != DNS_TYPE_DS)
+                return 0;
+
+        r = dns_answer_match_key(t->answer, dns_transaction_key(t), /* ret_flags = */ NULL);
+        if (r < 0)
+                return r;
+
```

Comment:
Please drop the unnecessary empty line. We usually do not insert any empty line between function call and error/result handling.